### PR TITLE
clean and tested

### PR DESCRIPTION
**Pull request by John-Colvin**
Once this is merged here I will initiate the transfer to dlangscience. You will also become a member of the DlangScience github organisation with push privileges to the nlopt repository, although we do ask that you make pull requests instead of pushing directly to master and don't merge your own pulls until another member has reviewed the request.

### CODE REVIEW

File: source/nlopt.d [hunk 1]
```diff
@@ -163,7 +163,7 @@ void nlopt_srand(ulong seed);
 void nlopt_srand_time();
 void nlopt_version(int* major, int* minor, int* bugfix);
 
-struct nlopt_opt_s{}; /* opaque structure, defined internally */
+struct nlopt_opt_s; /* opaque structure, defined internally */
```
Comment (John-Colvin) [inline source/nlopt.d]: Using a truly opaque struct means that it is a compile-time error for someone to try to use one directly (e.g. `nlopt_opt_s myopt;`), which is nice.

**Comment by John-Colvin** [comment]
by the way, some of the errors you may have had with that unittest could be down to https://issues.dlang.org/show_bug.cgi?id=15712

**Comment by John-Colvin** [comment]
I'm very busy at the moment, so it might be a while before I can do the same thing as this pull request for https://github.com/dataPulverizer/glpk, if you have time to do it then it would probably be quicker.

**Comment by dataPulverizer** [comment]
Great. I'll take a look at these.

**Comment by John-Colvin** [comment]
I've added you as a member of a team on DlangScience, I think you have to make the transfer yourself.

**Comment by dataPulverizer** [comment]
Great. I can get access after work and make the transfer myself.

**Comment by dataPulverizer** [comment]
Hi John,

I have tried transferring the nlopt repository and have been told that I
can't because I don't have admin rights to DlangScience.

On Tue, Feb 23, 2016 at 10:51 AM, Data Pulverizer <data.pulverizer@gmail.com
> wrote:

> Great. I can get access after work and make the transfer myself.

**Comment by John-Colvin** [comment]
try again now?

**Comment by dataPulverizer** [comment]
Apologies, I'll do some prep work for it now.

**Pull request merged**